Sources/Citadel/SFTP/Client/SFTPClient.swift

```diff
@@ -63,11 +69,8 @@ public final class SFTPClient {
     
     /// Returns a unique request ID for use in an SFTP message. Does _not_ register the ID for
     /// a response; that is handled by `sendRequest(_:)`.
-    internal func allocateRequestId() -> UInt32 {
-        defer {
-            self.nextRequestId &+= 1
-        }
-        return self.nextRequestId
+    @MainActor internal func allocateRequestId() -> UInt32 {
```

Comment:
I think we should steer away from using `@MainActor` here. I'd either like another (our own) globalActor, or preferrably to wrap the value in a `NIOLockedValueBox`

Reply:
If you're unable to make that change, I can work on it sometime later